src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -25,23 +28,61 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
     erroredFields = [];
     @api pageLevelErrorMessageList = [];
 
+    @wire(getRecord, { recordId: '$recordId', optionalFields: '$fieldNames'})
+    wiredGetRecordMethod({ error, data }) {
+        if (data) {
+            this.record = data;
+            this.handleGetTemplate();
+        } else if (error) {
+            console.error(JSON.stringify(error));
+        }
+    }
+
     connectedCallback() {
+        
         if (this.batchId) {
             // When the form is being used for Batch Gift Entry, the Form Template JSON
             // uses the @wire service below to retrieve the Template using the Template Id
             // stored on the Batch.
             return;
         }
 
+        // check if there is a record id in the url
+        this.recordId = getQueryParameters().c__recordId;
+
         GeFormService.getFormTemplate().then(response => {
             // read the template header info
             if(response !== null && typeof response !== 'undefined') {
                 const { formTemplate } = response;
+                const fieldMappings = response.fieldMappingSetWrapper.fieldMappingByDevName;
+
+                // get the target field names to be used by getRecord
+                this.fieldNames = getRecordFieldNames(formTemplate, fieldMappings);
                 this.initializeForm(formTemplate);
             }
         });
     }
 
+    handleGetTemplate = async () => {
```

Comment:
Are we duplicating calls to GeFormService.getFormTemplate()? I just noticed that is called in the connectedCallback() up above, but wasn't sure if that was unusual that the template looks like it is being retrieved in two different places